src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -1477,12 +1473,12 @@ static PyObject* NPySecObj_connect(NPySecObj* self, PyObject* args) {
         PyErr_SetString(PyExc_ValueError, "child connection end must be  0 or 1");
         return NULL;
     }
-    Py_INCREF(self);
     hoc_pushx(childend);
     hoc_pushx(parentx);
     nrn_pushsec(self->sec_);
     nrn_pushsec(parent->sec_);
     simpleconnectsection();
+    Py_INCREF(self);
```

Comment:
Risky, because we'd need to check that `simpleconnectsection` can't get `self` via global variables.

Reply:
Even if it did, it would be an outrageous mistake to decref (take ownership of self)

Reply:
@1uc I have rebased and kept only the minimal changes. The only discussion point is perhaps this one. I wouldn't think we have to protect for the wildes cases, which is what I consider in case `simpleconnectsection()` would find `self` from a global var AND Decref it.